### PR TITLE
Bug fixes

### PR DESCRIPTION
**Pull request by Quesos-Nachos**
Bot will no longer attempt to deserialize warning messages if the cat API is down. Commented some stuff. Changed error levels on log messages. Disposed of CTS token.

### CODE REVIEW

File: Modix/Modules/CatModule.cs [hunk 1]
```diff
@@ -40,6 +40,8 @@ public class CatModule : ModuleBase
 
             // Send the link
             await Context.Channel.SendMessageAsync(message);
+
+            cts.Dispose();
```
Comment (Cisien) [inline Modix/Modules/CatModule.cs]: shouldn't this be called in a finally block?

File: Modix/Services/Cat/CatService.cs [hunk 2]
```diff
@@ -16,6 +16,11 @@ public struct URL
             public string file { get; set; }
         }
 
+        /// <summary>
+        /// Gets the cat GIF.
+        /// </summary>
+        /// <param name="token">The token.</param>
+        /// <returns>Return the URL to the cat</returns>
```
Comment (Cisien) [inline Modix/Services/Cat/CatService.cs]: I smell stock StyleCop comments -- my opinion on these is to either update them to be more useful than the method name and parameters suggest, or remove them.

File: Modix/Services/Cat/CatService.cs [hunk 3]
```diff
@@ -55,10 +74,19 @@ public struct URL
...
+                    Log.Warning("Could not retrieve JSON string within the alloted time. Is the API Down?");
+                    return "Cat not downloaded in time";
+                }
 
-                // Check and make sure the string isn't empty before attempting to deserialize cat
+                // Check and make sure the string isn't empty before attempting to deserialize the
+                // json. If the website returns a blank string, something is wrong. Break the loop
                 if (string.IsNullOrWhiteSpace(json)) break;
 
                 //Deserialize the json retrieved from the website
```
Comment (Cisien) [inline Modix/Services/Cat/CatService.cs]: I'm seeing doubles (this code is identical to the code above, and can probably be extracted to a common method

File: Modix/Services/Cat/CatService.cs [hunk 4]
```diff
@@ -79,7 +107,13 @@ public struct URL
             return fileFoundFlag ? obj.file : "404 cat picture not found";
         }
```
Comment (Cisien) [inline Modix/Services/Cat/CatService.cs]: I'm seeing doubles (this code is identical to the code above, and can probably be extracted to a common method

File: Modix/Modules/CatModule.cs (hunk 1 above)
Comment (Cisien) [reply Modix/Modules/CatModule.cs]: or a using

**Comment by Quesos-Nachos** [comment]
I'm going to rewrite some of it to use Discord's built in dependency injection per Inzanit's suggestion. Going to add an interface and some other stuff.

File: Modix/Modules/CatModule.cs [hunk 5]
```diff
@@ -3,39 +3,51 @@
...
-            // Regular picture
-            if (!string.Equals("gif", gif, StringComparison.OrdinalIgnoreCase))
+            try
             {
-                message = await cat.GetCatPicture(token);
-            }
-            else if (gif.Equals("gif", StringComparison.OrdinalIgnoreCase))
+                MediaType = (Media) Enum.Parse(typeof(Media), param, true);
+
+                if (!Enum.IsDefined(typeof(Media), MediaType) && !MediaType.ToString().Contains(","))
```
Comment (Cisien) [inline Modix/Modules/CatModule.cs]: if enum is not defined, and it doesn't contain a comma? An enum would never contain a comma...?

File: Modix/Services/Cat/CatService.cs [hunk 6]
```diff
@@ -5,81 +5,68 @@
...
 
-    public class CatService
+    public interface ICatService
+    {
+        Task<string> HandleCat(CatModule.Media mediaType, CancellationToken token);
+    }
+
+    public class CatService : ICatService
     {
         private static readonly HttpClient Client = new HttpClient();
```
Comment (Cisien) [inline Modix/Services/Cat/CatService.cs]: I'd make this a private readonly member, and register the service as a singleton instead

**Comment by Quesos-Nachos** [comment]
Registered service as Singleton, moved the enum, and removed the enum parsing logic. (I didn't like it. I think the **if** statements are easier to read)  The command can take a Media type parameter, but each command has to be either **!cat picture** or **!cat gif**. I wanted to retain functionality for **!cat** being the command to get pictures

**Pull request merged**